src/main/java/org/oscii/detokenize/TrainDetokenizer.java

```diff
@@ -0,0 +1,103 @@
+package org.oscii.detokenize;
+
+import com.google.common.collect.Iterators;
+import edu.stanford.nlp.mt.process.Preprocessor;
+import edu.stanford.nlp.mt.process.de.GermanPreprocessor;
+import edu.stanford.nlp.mt.process.en.EnglishPreprocessor;
+import edu.stanford.nlp.mt.process.es.SpanishPreprocessor;
+import edu.stanford.nlp.mt.process.fr.FrenchPreprocessor;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Command-line utility to train a detokenizer.
+ */
+public class TrainDetokenizer {
+    public static void main(String[] args) throws IOException {
+        OptionSet options = parse(args);
+
+        File trainFile = (File) options.valueOf("data");
+        InputStream trainStream = new FileInputStream(trainFile);
+        if (trainFile.getName().endsWith(".gz")) {
+            trainStream = new GZIPInputStream(trainStream);
+        }
+        BufferedReader buffered = new BufferedReader(new InputStreamReader(trainStream, "utf-8"));
+        Iterator<String> trainExamples = buffered.lines().iterator();
+
+        int testSize = (Integer) options.valueOf("testsize");
+        List<String> testExamples = new ArrayList<>(testSize);
+        for (int i = 0; i < testSize; i++) {
+            testExamples.add(trainExamples.next());
```

Comment:
doesn't matter for the data that I provided which was a random sample, but it would be safer to extract a random test set from the training data in case the input data is non-randomized, e.g. sorted alphanumerically or by input length